tests/keras_tests/pruning_tests/test_models.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+
+from model_compression_toolkit.core.common.pruning.importance_metrics.importance_metric_factory import im_dict
+from tests.keras_tests.pruning_tests.random_importance_metric import RandomImportanceMetric
+
+keras = tf.keras
+layers = keras.layers
+
+NUM_PRUNING_RATIOS = 2
+
+class TestImportanceMetric(Enum):
+    RANDOM = 'random'
+
+im_dict.update({TestImportanceMetric.RANDOM: RandomImportanceMetric})
+
+class ModelsPruningTest(unittest.TestCase):
+    def representative_dataset(self, in_shape=(1,224,224,3)):
+        for _ in range(1):
+            yield [np.random.randn(*in_shape)]
+
+    def test_rn50_pruning(self):
+        from keras.applications.resnet50 import ResNet50
+        dense_model = ResNet50()
+        target_crs = np.linspace(0.5, 1, NUM_PRUNING_RATIOS)
+        for cr in target_crs:
+            self.run_test(cr, dense_model)
+
+    def test_efficientnetb0_pruning(self):
+        from keras.applications.efficientnet import EfficientNetB0
+        dense_model = EfficientNetB0()
+        target_crs = np.linspace(0.8, 1, NUM_PRUNING_RATIOS)
+        for cr in target_crs:
+            self.run_test(cr, dense_model)
+
+
+    def test_vgg16_pruning(self):
+        from keras.applications.vgg16 import VGG16
+        dense_model = VGG16()
+        target_crs = np.linspace(0.5, 1, NUM_PRUNING_RATIOS)
+        for cr in target_crs:
+            self.run_test(cr, dense_model)
+
+
+    def test_mobilenet_pruning(self):
+        from keras.applications.mobilenet import MobileNet
+        dense_model = MobileNet()
+        target_crs = np.linspace(0.55, 1, NUM_PRUNING_RATIOS)
+        for cr in target_crs:
+            self.run_test(cr, dense_model)
+
+    def test_mobilenetv2_pruning(self):
+        from keras.applications.mobilenet_v2 import MobileNetV2
+        dense_model = MobileNetV2()
+        target_crs = np.linspace(0.5, 1, NUM_PRUNING_RATIOS)
+        for cr in target_crs:
+            self.run_test(cr, dense_model)
+
+    def test_densenet_pruning(self):
+        from keras.applications.densenet import DenseNet121
+        dense_model = DenseNet121()
+        target_crs = np.linspace(0.5, 1, NUM_PRUNING_RATIOS)
+        for cr in target_crs:
+            self.run_test(cr, dense_model)
+
+
+    def test_vgg19_pruning(self):
+        from keras.applications.vgg19 import VGG19
+        dense_model = VGG19()
+        target_crs = np.linspace(0.5, 1, NUM_PRUNING_RATIOS)
+        for cr in target_crs:
+            self.run_test(cr, dense_model)
+
+
+    def _dummy_retrain(self, model, ds):
+        # Compile the model with a loss function, optimizer, and metric to monitor
+        model.compile(optimizer='adam',
+                      loss='sparse_categorical_crossentropy',
+                      metrics=['accuracy'])
+
+        # Train the model for one epoch using the dummy dataset
+        model.fit(ds, epochs=1)
+        return model
+
+    def run_test(self, cr, dense_model, test_retraining=False):
```

Comment:
please add some comments explaining the test parts... it's not clear what is actually being tested